frontend/src/workspace/funbar/FunBar.js

```diff
@@ -0,0 +1,187 @@
+          native
+          from={{
+            fill: "orange",
+          }}
+          to={{
+            fill:
+              props.renderFunction.isRenderable && functionButtonHovered
+                ? "white"
+                : "orange",
+          }}
+        >
+          {(props) => (
+            <animated.Rect // function button
+              {...props}
+              x={0}
+              y={0}
+              width={funbarDimensions.functionButtonWidth}
+              height={funbarDimensions.functionButtonHeight}
+              //stroke={"#424874"}
+              cornerRadius={8}
+              shadowBlur={5}
+              shadowOffset={{ x: 2, y: 3 }}
+              shadowOpacity={0.5}
+              opacity={renderFunction.isRenderable ? 1 : 0.8}
+            />
+          )}
+        </Spring>
+        <Text // function button
+          text={"Function"}
+          x={0}
+          width={funbarDimensions.functionButtonWidth}
+          height={funbarDimensions.functionButtonHeight}
+          align={"center"}
+          verticalAlign={"middle"}
+          fill={!functionButtonHovered ? "white" : "grey"}
+          fontSize={funBarFontSize}
+          onMouseOver={() => {
+            setFunctionButtonHovered(true);
+          }}
+          onMouseOut={() => {
+            setFunctionButtonHovered(false);
+          }}
+        />
+      </Group>
+    );
+  }
+
+  function ImageButton(props) {
+    function closePortal() {
+      setImageButtonClicked(false);
+    }
+
+    return (
+      <Group // Image Button on blue bar
+        x={funbarDimensions.imageButtonX}
+        y={funbarDimensions.margin}
+      >
+        <Spring // animates image button fill
+          native
+          from={{
+            fill: "orange",
+          }}
+          to={{
+            fill:
+              props.renderFunction.isRenderable && imageButtonHovered
+                ? "white"
+                : "orange",
+          }}
+        >
+          {(props) => (
+            <animated.Rect // function button
+              {...props}
+              width={funbarDimensions.imageButtonWidth}
+              height={funbarDimensions.imageButtonHeight}
+              //stroke={"#424874"}
+              cornerRadius={8}
+              shadowBlur={5}
+              shadowOffset={{ x: 2, y: 3 }}
+              shadowOpacity={0.5}
+              opacity={renderFunction.isRenderable ? 1 : 0.8}
+            />
+          )}
+        </Spring>
+        <Text
+          text={"Image"}
+          width={funbarDimensions.imageButtonWidth}
+          height={funbarDimensions.imageButtonHeight}
+          align={"center"}
+          verticalAlign={"middle"}
+          fill={imageButtonHovered ? "gray" : "white"}
+          fontSize={funBarFontSize}
+          onClick={() => {
+            if (props.renderFunction.isRenderable) {
+              //setImageButtonClicked(true);
+              setImageButtonHovered(false);
```

Comment:
Line 151 and line 170. Are these being left as an example or are they no longer necessary? If the latter, then the commented line should be removed.